### ISSUE TITLE
Compatibility with docker exec when piping

### ISSUE THREAD

**Issue by dpatel257** [hadolint/hadolint#426]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

This is defaut usage mentioned in README:
`docker run -i --name my_linter hadolint/hadolint < Dockerfile`

this works fine. But with `docker exec` I am not able to pipe files
```
docker run -di --name my_linter hadolint/hadolint
docker exec -i my_linter hadolint/hadolint < Dockerfile
OCI runtime exec failed: exec failed: container_linux.go:346: starting container process caused "exec: \"hadolint/hadolint\": stat hadolint/hadolint: no such file or directory": unknown

```

**Comment by zemanlx** [comment]
Hi @dpatel257,

IMHO you do not call it correctly. `hadolint/hadolint` is the name of hadolint image not an executable. If you want to pass Dockerfile to hadolint during exec you can call it like this

```bash
docker exec -i my_linter hadolint - < Dockerfile
```

Let me know if it helps.

**Comment by dpatel257** [comment]
Thanks for your quick reply.
That worked like a charm. Appreciate it.
